packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -41,7 +41,7 @@ const String threeLines = 'line1\nline2\nline3';
 Widget buildInputDecorator({
```

Comment:
Can we add a test to check if a local InputDecorationTheme can override the themedata.inputDecortationTheme?

Reply:
Oh, I missed that. It will required several changes as the current implementation as many references to `Theme.of(context).inputDecorationTheme`.
For the test, I will create a group because there are many properties and combinations to test.

Reply:
Thank you! 

> Some properties are missing because there are no M3 tests for them. To make it easier to track this work, I will filed other PRs to add the missing M3 tests

We don't need to add the M3 tests if M3 didn't specify the property in defaults. It would be fine as long as we can make sure the properties works when we use them in a local `InputDecorationThemeData`. 

> when needed, fix the implementation if it does not use the local theme properly.

Does this mean that some properties don't work in a `InputDecorationThemeData`? If so, maybe we should not add them in `InputDecorationThemeData`.

Reply:
Migrating InputDecoration tests to M3 is still a work in a progress. This is why some properties are missing existing M3 tests. 
To correctly check that the local theme properly work for those properties I would first have to add tests that check if the general theme is working for those properties (hopefully most of them are probably ok, otherwise we would have get some reports).
My problem is that I prefer to do that in separate PRs as it becomes difficult to track changes in this PR, and It would be more revert proof.

Reply:
Ah I see. Yeah, separating PRs makes sense to me!